pkg/cmd/pipeline/translate.go

```diff
@@ -0,0 +1,226 @@
+			return nil, fmt.Errorf("Another pipeline operation is being executed")
+		}
+		return nil, err
+	}
+	return cmap, nil
+}
+
+// SetOutput sets the output of a config map
+func SetOutput(cfg *apiv1.ConfigMap, output string) *apiv1.ConfigMap {
+	cfg.Data[outputField] = base64.StdEncoding.EncodeToString([]byte(output))
+	return cfg
+}
+
+// UpdateConfigMap updates the configmaps fields
+func UpdateConfigMap(ctx context.Context, cmap *apiv1.ConfigMap, data *CfgData, c kubernetes.Interface) error {
+	cmap, err := configmaps.Get(ctx, cmap.Name, cmap.Namespace, c)
+	if err != nil {
+		return err
+	}
+	if err := updateCmap(cmap, data); err != nil {
+		return err
+	}
+	return configmaps.Deploy(ctx, cmap, cmap.Namespace, c)
+}
+
+// TranslatePipelineName translate the name into the pipeline name
+func TranslatePipelineName(name string) string {
+	return fmt.Sprintf("okteto-git-%s", name)
+}
+
+func translateOutput(output *bytes.Buffer) []byte {
+	// If the output is larger than the currentMaxLimit for the logs trim it.
+	// We can't really truncate the buffer since we would end up with an invalid json
+	// line for the last line, so we pick lines from the end while the line fits
+	var data []byte
+	if output.Len() > maxLogOutputRaw {
+		scanner := bufio.NewScanner(output)
+		linesInReverse := []string{}
+		for scanner.Scan() {
+			linesInReverse = append([]string{scanner.Text()}, linesInReverse...)
+		}
+		var cappedOutput string
+		var head string
+		for _, l := range linesInReverse {
+			head = l + "\n"
+			if len(cappedOutput)+len(head) > maxLogOutputRaw {
+				break
+			}
+			cappedOutput = head + cappedOutput
+		}
+		cappedOutput = strings.TrimPrefix(cappedOutput, head)
+		data = []byte(cappedOutput)
+	} else {
+		data = output.Bytes()
+	}
+	return data
+}
+
+func translatedConfigMapSandBox(data *CfgData) *apiv1.ConfigMap {
+	cmap := &apiv1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: data.Namespace,
+			Name:      TranslatePipelineName(data.Name),
+			Labels: map[string]string{
+				model.GitDeployLabel: "true",
+			},
+		},
+		Data: map[string]string{
+			nameField:     data.Name,
+			statusField:   data.Status,
+			repoField:     data.Repository,
+			branchField:   data.Branch,
+			filenameField: data.Filename,
+			yamlField:     base64.StdEncoding.EncodeToString(data.Manifest),
+			iconField:     data.Icon,
+		},
+	}
+	if data.Repository != "" {
+		cmap.Data[repoField] = data.Repository
+	}
+
+	if data.Branch != "" {
+		cmap.Data[branchField] = data.Branch
+	}
+
+	output := oktetoLog.GetOutputBuffer()
+	outputData := translateOutput(output)
+	cmap.Data[outputField] = base64.StdEncoding.EncodeToString([]byte(outputData))
+	return cmap
+}
+
+func updateCmap(cmap *apiv1.ConfigMap, data *CfgData) error {
+	actionName := os.Getenv(model.OktetoActionNameEnvVar)
+	if actionName == "" {
+		actionName = actionDefaultName
+	}
+	if _, ok := cmap.Data[actionLockField]; ok {
+		return errors.New("There is a deploy operation already running")
```

Comment:
Is this the final error shown to the user? If so, I think it is not accurate. Potentially, you can execute 2 deploy operations in the same folder from 2 different terminals (nothing is blocking that AFAIK), what you cannot execute is a `pipeline` and a `deploy` for the same pipeline/application.